sympy/polys/matrices/tests/test_ddm.py

```diff
@@ -556,3 +557,15 @@ def test_DDM_is_lower():
     ], (4, 3), QQ).transpose()
     assert A.is_lower() is True
     assert B.is_lower() is False
+
+
+def test_ddm_qr():
+    # Create a sample matrix in DomainMatrix
+    A = DomainMatrix([[ZZ(3), ZZ(1)], [ZZ(4), ZZ(3)]], (2, 2), ZZ)
```

Comment:
The tests for this should test all implementations as in:
https://github.com/sympy/sympy/blob/e53690af7e57eeef7e8c6dea1521a4dc0024cf95/sympy/polys/matrices/tests/test_xxm.py#L829-L845